src/main/groovy/lang/IntRange.java

```diff
@@ -385,26 +385,23 @@ public void step(int step, Closure closure) {
             return; // from == to and step == 0, nothing to do, so return
         }
 
-        if (isReverse()) {
-            step = -step;
+        final long from = getFrom().longValue(), to = getTo().longValue();
+
+        final boolean isAscending;
+        if (step < 0) {
+            step *= -1;
+            isAscending = isReverse();
+        } else {
+            isAscending = !isReverse();
         }
-        if (step > 0) {
-            int value = getFrom();
-            while (value <= getTo()) {
+
+        if (isAscending) {
```

Comment:
its easier to think about ascending and descending iterations than with bigger from values with negative step sizes...